### DISCUSSION TITLE
WebSocket TLS

### DISCUSSION THREAD

**Discussion by sakoula** [dani-garcia/vaultwarden#1424] [Q&A]
Hi,

Thanks for the great software!!!!

Quick question. `WebSocket notifications` [support](https://github.com/dani-garcia/bitwarden_rs/wiki/Enabling-WebSocket-notifications) works only for http/clear text (port 3012).

Can I somehow enable TLS/SSL for this port?

Thanks!

PS: I just found the discourge group. I should have posted that there instead of here. Sorry!

**Comment by BlackDex** [comment]
For websockets you need a reverse proxy where you (can) should offload the ssl which then also enables that on the websockets.

**Reply by sakoula** [reply]
hi thanks for the reply. Yes this is how I did it. I wanted to encrypt the connection between the reverse proxy and the websockets server. This does not seem to be doable. is there another way to do it? thanks!

**Reply by mqus** [reply]
As a workaround, you could try to set up a forwarded port via ssh and add firewall rules to only make the ports accessible from localhost each.

Me personally, I just run bitwarden_rs on the same machine as the reverse proxy.
